<?xml version="1.0" encoding="UTF-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro4sdf from model.sdf.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<sdf version="1.7">
	<model name="standard_robot_blue">
		<pose>0 0 0.15 0 0 0</pose>
		<!--chassis-->
		<link name="chassis">
			<inertial>
				<mass>10.0</mass>
				<inertia>
					<ixx>0.06675</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.12675</iyy>
					<iyz>0</iyz>
					<izz>0.156</izz>
				</inertia>
			</inertial>
			<visual name="chassis_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/chassis_base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="chassis_base_collision">
				<pose>0 0 0.055 0 0 0</pose>
				<geometry>
					<box>
						<size>0.36 0.24 0.15</size>
					</box>
				</geometry>
			</collision>
			<collision name="chassis_projectile_container_collision">
				<pose>-0.122 0 0.177 0 0 0</pose>
				<geometry>
					<box>
						<size>0.1 0.18 0.146</size>
					</box>
				</geometry>
			</collision>
			<collision name="chassis_fornt_wheel_holder_collision">
				<pose>0.208 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.055 0.367 0.055</size>
					</box>
				</geometry>
			</collision>
			<collision name="chassis_rear_wheel_holder_collision">
				<pose>-0.208 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.055 0.367 0.055</size>
					</box>
				</geometry>
			</collision>
			<sensor name="chassis_imu" type="imu">
				<always_on>1</always_on>
				<update_rate>30</update_rate>
				<visualize>0</visualize>
			</sensor>
		</link>
		<!--light_indicator-->
		<joint name="light_indicator_joint" type="fixed">
			<pose relative_to="chassis">-0.207 0 0.1 0 0 0</pose>
			<parent>chassis</parent>
			<child>light_indicator</child>
		</joint>
		<link name="light_indicator">
			<pose relative_to="light_indicator_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0.04 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>0.0006666666666666669</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>3.416666666666667e-05</iyy>
					<iyz>0</iyz>
					<izz>0.0006741666666666668</izz>
				</inertia>
			</inertial>
			<visual name="light_indicator_holder_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/light_indicator_holder.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="light_indicator_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/light_indicator_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="light_indicator_bar_collision">
				<pose>0 0 0.06 0 0 0</pose>
				<geometry>
					<box>
						<size>0.05 0.28 0.04</size>
					</box>
				</geometry>
			</collision>
		</link>
		<!--front_left wheel link and joint-->
		<joint name="front_left_wheel_joint" type="revolute">
			<pose relative_to="chassis">0.207 0.194 0 0 0 0</pose>
			<parent>chassis</parent>
			<child>front_left_wheel</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-1.79769e+308</lower>
					<upper>1.79769e+308</upper>
				</limit>
			</axis>
		</joint>
		<link name="front_left_wheel">
			<pose relative_to="front_left_wheel_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0 1.5707 0 0</pose>
				<mass>0.4</mass>
				<inertia>
					<ixx>0.0006426653333333334</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0006426653333333334</iyy>
					<iyz>0</iyz>
					<izz>0.0011491280000000001</izz>
				</inertia>
			</inertial>
			<visual name="front_left_wheel_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/wheel_left.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="front_left_wheel_collision">
				<pose>0 0 0 1.5707 0 0</pose>
				<geometry>
					<cylinder>
						<radius>0.0758</radius>
						<length>0.0452</length>
					</cylinder>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>0.2</mu>
							<mu2>0.2</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
		</link>
		<!--front_right wheel link and joint-->
		<joint name="front_right_wheel_joint" type="revolute">
			<pose relative_to="chassis">0.207 -0.194 0 0 0 0</pose>
			<parent>chassis</parent>
			<child>front_right_wheel</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-1.79769e+308</lower>
					<upper>1.79769e+308</upper>
				</limit>
			</axis>
		</joint>
		<link name="front_right_wheel">
			<pose relative_to="front_right_wheel_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0 1.5707 0 0</pose>
				<mass>0.4</mass>
				<inertia>
					<ixx>0.0006426653333333334</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0006426653333333334</iyy>
					<iyz>0</iyz>
					<izz>0.0011491280000000001</izz>
				</inertia>
			</inertial>
			<visual name="front_right_wheel_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/wheel_right.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="front_right_wheel_collision">
				<pose>0 0 0 1.5707 0 0</pose>
				<geometry>
					<cylinder>
						<radius>0.0758</radius>
						<length>0.0452</length>
					</cylinder>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>0.2</mu>
							<mu2>0.2</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
		</link>
		<!--rear_left wheel link and joint-->
		<joint name="rear_left_wheel_joint" type="revolute">
			<pose relative_to="chassis">-0.207 0.194 0 0 0 0</pose>
			<parent>chassis</parent>
			<child>rear_left_wheel</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-1.79769e+308</lower>
					<upper>1.79769e+308</upper>
				</limit>
			</axis>
		</joint>
		<link name="rear_left_wheel">
			<pose relative_to="rear_left_wheel_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0 1.5707 0 0</pose>
				<mass>0.4</mass>
				<inertia>
					<ixx>0.0006426653333333334</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0006426653333333334</iyy>
					<iyz>0</iyz>
					<izz>0.0011491280000000001</izz>
				</inertia>
			</inertial>
			<visual name="rear_left_wheel_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/wheel_right.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="rear_left_wheel_collision">
				<pose>0 0 0 1.5707 0 0</pose>
				<geometry>
					<cylinder>
						<radius>0.0758</radius>
						<length>0.0452</length>
					</cylinder>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>0.2</mu>
							<mu2>0.2</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
		</link>
		<!--rear_right wheel link and joint-->
		<joint name="rear_right_wheel_joint" type="revolute">
			<pose relative_to="chassis">-0.207 -0.194 0 0 0 0</pose>
			<parent>chassis</parent>
			<child>rear_right_wheel</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-1.79769e+308</lower>
					<upper>1.79769e+308</upper>
				</limit>
			</axis>
		</joint>
		<link name="rear_right_wheel">
			<pose relative_to="rear_right_wheel_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0 1.5707 0 0</pose>
				<mass>0.4</mass>
				<inertia>
					<ixx>0.0006426653333333334</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0006426653333333334</iyy>
					<iyz>0</iyz>
					<izz>0.0011491280000000001</izz>
				</inertia>
			</inertial>
			<visual name="rear_right_wheel_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/wheel_left.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="rear_right_wheel_collision">
				<pose>0 0 0 1.5707 0 0</pose>
				<geometry>
					<cylinder>
						<radius>0.0758</radius>
						<length>0.0452</length>
					</cylinder>
				</geometry>
				<surface>
					<friction>
						<ode>
							<mu>0.2</mu>
							<mu2>0.2</mu2>
						</ode>
					</friction>
				</surface>
			</collision>
		</link>
		<!--4 armors -->
		<!--front armors -->
		<joint name="armor_0_joint" type="fixed">
			<pose relative_to="chassis">0.197 0.0 0.061 0 0 0.0</pose>
			<parent>chassis</parent>
			<child>armor_0</child>
		</joint>
		<link name="armor_0">
			<pose relative_to="armor_0_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.04 0 0 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>0.0003065083333333333</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00020463541666666669</iyy>
					<iyz>0</iyz>
					<izz>0.00022947708333333333</izz>
				</inertia>
			</inertial>
			<visual name="armor_0_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="armor_0_light_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_light_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="armor_0_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--rear armors -->
		<joint name="armor_2_joint" type="fixed">
			<pose relative_to="chassis">-0.197 0.0 0.061 0 0 3.14159</pose>
			<parent>chassis</parent>
			<child>armor_2</child>
		</joint>
		<link name="armor_2">
			<pose relative_to="armor_2_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.04 0 0 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>0.0003065083333333333</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00020463541666666669</iyy>
					<iyz>0</iyz>
					<izz>0.00022947708333333333</izz>
				</inertia>
			</inertial>
			<visual name="armor_2_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="armor_2_light_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_light_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="armor_2_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--left armors -->
		<joint name="armor_1_joint" type="fixed">
			<pose relative_to="chassis">0.0 0.116 0.06 0 0 1.5708</pose>
			<parent>chassis</parent>
			<child>armor_1</child>
		</joint>
		<link name="armor_1">
			<pose relative_to="armor_1_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.04 0 0 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>0.0003065083333333333</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00020463541666666669</iyy>
					<iyz>0</iyz>
					<izz>0.00022947708333333333</izz>
				</inertia>
			</inertial>
			<visual name="armor_1_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="armor_1_light_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_light_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="armor_1_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--right armors -->
		<joint name="armor_3_joint" type="fixed">
			<pose relative_to="chassis">0.0 -0.116 0.06 0 0 -1.5708</pose>
			<parent>chassis</parent>
			<child>armor_3</child>
		</joint>
		<link name="armor_3">
			<pose relative_to="armor_3_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.04 0 0 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>0.0003065083333333333</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00020463541666666669</iyy>
					<iyz>0</iyz>
					<izz>0.00022947708333333333</izz>
				</inertia>
			</inertial>
			<visual name="armor_3_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="armor_3_light_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_light_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="armor_3_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/armor_base.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--gimbal yaw module-->
		<joint name="gimbal_yaw_joint" type="revolute">
			<pose relative_to="chassis">0 0 0.1376 0 0 0</pose>
			<parent>chassis</parent>
			<child>gimbal_yaw</child>
			<axis>
				<xyz>0 0 1</xyz>
				<limit>
					<lower>-1.57</lower>
					<upper>1.57</upper>
				</limit>
			</axis>
		</joint>
		<link name="gimbal_yaw">
			<pose relative_to="gimbal_yaw_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0.06 0 0 0</pose>
				<mass>0.5</mass>
				<inertia>
					<ixx>0.001516666666666667</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.001516666666666667</iyy>
					<iyz>0</iyz>
					<izz>0.0006250000000000001</izz>
				</inertia>
			</inertial>
			<visual name="gimbal_yaw_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/gimbal_yaw.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="gimbal_yaw_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/gimbal_yaw.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--gimbal pitch module-->
		<joint name="gimbal_pitch_joint" type="revolute">
			<pose relative_to="gimbal_yaw">0 0.004 0.16 0 0 0</pose>
			<parent>gimbal_yaw</parent>
			<child>gimbal_pitch</child>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-0.785</lower>
					<upper>0.5652</upper>
				</limit>
			</axis>
		</joint>
		<link name="gimbal_pitch">
			<pose relative_to="gimbal_pitch_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.05 0 0 0 0 0</pose>
				<mass>0.2</mass>
				<inertia>
					<ixx>0.0003016666666666667</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.0012083333333333336</iyy>
					<iyz>0</iyz>
					<izz>0.0011766666666666668</izz>
				</inertia>
			</inertial>
			<visual name="gimbal_pitch_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/gimbal_pitch.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="gimbal_pitch_collision">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/gimbal_pitch.dae</uri>
					</mesh>
				</geometry>
			</collision>
		</link>
		<!--speed_monitor_17mm-->
		<joint name="speed_monitor_17mm_joint" type="fixed">
			<pose relative_to="gimbal_pitch">0.07 0 0 0 0 0</pose>
			<parent>gimbal_pitch</parent>
			<child>speed_monitor_17mm</child>
		</joint>
		<link name="speed_monitor_17mm">
			<pose relative_to="speed_monitor_17mm_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.055 0 0 0 0 0</pose>
				<mass>0.1</mass>
				<inertia>
					<ixx>2.941666666666667e-05</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>0.00011484166666666668</iyy>
					<iyz>0</iyz>
					<izz>0.00011624166666666668</izz>
				</inertia>
			</inertial>
			<visual name="base_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/speed_monitor.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<visual name="light_bar_visual">
				<geometry>
					<mesh>
						<uri>model://rmua19_standard_robot/meshes/speed_monitor_light_bar.dae</uri>
					</mesh>
				</geometry>
				<material>
					<ambient>0 0 1 1</ambient>
					<diffuse>0 0 1 1</diffuse>
					<specular>1 1 1 1</specular>
					<emissive>0 0 1 1</emissive>
				</material>
			</visual>
			<collision name="base_collision">
				<pose>0.055 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.11 0.043 0.041</size>
					</box>
				</geometry>
			</collision>
		</link>
		<!--rplidar a2-->
		<joint name="front_rplidar_a2_joint" type="fixed">
			<pose relative_to="chassis">0.155 0 0.1 0 0 0</pose>
			<parent>chassis</parent>
			<child>front_rplidar_a2</child>
		</joint>
		<link name="front_rplidar_a2">
			<pose relative_to="front_rplidar_a2_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0 0 0.02 0 0 0</pose>
				<mass>0.005</mass>
				<inertia>
					<ixx>2.6666666666666664e-06</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>2.6666666666666664e-06</iyy>
					<iyz>0</iyz>
					<izz>4.000000000000001e-06</izz>
				</inertia>
			</inertial>
			<visual name="visual">
				<geometry>
					<mesh>
						<uri>model://rplidar_a2/meshes/rplidar_a2.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="collision">
				<pose>0 0 0.02 0 0 0</pose>
				<geometry>
					<cylinder>
						<radius>0.04</radius>
						<length>0.04</length>
					</cylinder>
				</geometry>
			</collision>
			<sensor name="front_rplidar_a2" type="gpu_lidar">
				<pose>0 0 0.03 0 0 0</pose>
				<always_on>true</always_on>
				<visualize>true</visualize>
				<update_rate>10.0</update_rate>
				<ray>
					<scan>
						<horizontal>
							<samples>400.0</samples>
							<resolution>0.9</resolution>
							<min_angle>-3.14159265359</min_angle>
							<max_angle>3.14159265359</max_angle>
						</horizontal>
					</scan>
					<range>
						<min>0.15</min>
						<max>12.0</max>
					</range>
				</ray>
				<noise>
					<type>gaussian</type>
					<mean>0.0</mean>
					<stddev>0.01</stddev>
				</noise>
			</sensor>
		</link>
		<!--industrial camera-->
		<joint name="front_industrial_camera_joint" type="fixed">
			<pose relative_to="gimbal_pitch">0.1 0 0.045 0 0 0</pose>
			<parent>gimbal_pitch</parent>
			<child>front_industrial_camera</child>
		</joint>
		<link name="front_industrial_camera">
			<pose relative_to="front_industrial_camera_joint">0 0 0 0 0 0</pose>
			<inertial>
				<pose>0.025 0 0 0 0 0</pose>
				<mass>0.05</mass>
				<inertia>
					<ixx>7.141666666666669e-06</ixx>
					<ixy>0</ixy>
					<ixz>0</ixz>
					<iyy>3.1204166666666675e-05</iyy>
					<iyz>0</iyz>
					<izz>2.9270833333333334e-05</izz>
				</inertia>
			</inertial>
			<visual name="visual">
				<geometry>
					<mesh>
						<uri>model://industrial_camera/meshes/industrial_camera.dae</uri>
					</mesh>
				</geometry>
			</visual>
			<collision name="collision">
				<pose>0.025 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.08 0.025 0.033</size>
					</box>
				</geometry>
			</collision>
			<sensor name="front_industrial_camera" type="camera">
				<always_on>1</always_on>
				<update_rate>30.0</update_rate>
				<camera>
					<horizontal_fov>1.0</horizontal_fov>
					<image>
						<width>640.0</width>
						<height>480.0</height>
					</image>
					<clip>
						<near>0.1</near>
						<far>100</far>
					</clip>
				</camera>
			</sensor>
		</link>
		<!--plugins-->
		<!--plugin for joint state sensor-->
		<plugin filename="ignition-gazebo-joint-state-publisher-system" name="ignition::gazebo::systems::JointStatePublisher">
			<joint_name>gimbal_yaw_joint</joint_name>
			<joint_name>gimbal_pitch_joint</joint_name>
		</plugin>
		<!--plugin for chassis control-->
		<plugin filename="MecanumDrive" name="ignition::gazebo::systems::MecanumDrive">
			<chassis_link>chassis</chassis_link>
			<front_left_joint>front_left_wheel_joint</front_left_joint>
			<front_right_joint>front_right_wheel_joint</front_right_joint>
			<rear_left_joint>rear_left_wheel_joint</rear_left_joint>
			<rear_right_joint>rear_right_wheel_joint</rear_right_joint>
		</plugin>
		<!--plugin for gimbal control-->
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>gimbal_yaw_joint</joint_name>
			<p_gain>0.5</p_gain>
			<i_gain>0.01</i_gain>
			<d_gain>0.1</d_gain>
			<i_max>1</i_max>
			<i_min>-1</i_min>
			<cmd_max>30</cmd_max>
			<cmd_min>-30</cmd_min>
		</plugin>
		<plugin filename="ignition-gazebo-joint-position-controller-system" name="ignition::gazebo::systems::JointPositionController">
			<joint_name>gimbal_pitch_joint</joint_name>
			<p_gain>2.0</p_gain>
			<i_gain>0.1</i_gain>
			<d_gain>0.2</d_gain>
			<i_max>10</i_max>
			<i_min>-10</i_min>
			<cmd_max>50</cmd_max>
			<cmd_min>-50</cmd_min>
		</plugin>
	</model>
</sdf>
